measureme/src/event_id.rs

```diff
@@ -78,4 +78,17 @@ impl<'p> EventIdBuilder<'p> {
             StringComponent::Ref(arg),
         ]))
     }
+
+    pub fn from_label_and_args(&self, label: StringId, args: &[StringId]) -> EventId {
+        // The capacity is the number of args + the same number for arg separators + the label.
+        let mut parts = Vec::with_capacity(args.len() * 2 + 1);
```

Comment:
Looks like a good use case for a `SmallVec<[StringComponent; 8]>`. I'd be OK with adding a dependency on https://crates.io/crates/smallvec.

Reply:
Done! It was already in our dependency tree anyway so it shouldn't hurt our build times. 

I went ahead and used `7` instead of `8` since there's no situation where the `smallvec` will have 8 items in it.

Reply:
:+1: